src/tactic/ui/widget/sobject_calendar_wdg.py

```diff
@@ -1319,6 +1349,107 @@ def get_day_wdg(my, month, day):
         line_div.add("%s work hours<br/>" % work_hours)
 
 
+
+
+
+
+
+
+
+
+
+
+        line_div = DivWdg()
+        div.add(line_div)
+        line_div.add_style("padding: 3px")
+
+
+
+        num_milestone = my.milestones_count.get(key)
+        if num_milestone:
+            icon = IconWdg("Milestones", IconWdg.GOOD)
+            line_div.add(icon)
+            line_div.add_style("font-style: italic")            
+            line_div.add_style("cursor: pointer;")
+            line_div.add("Has %i milestone(s)<br/>" % (num_milestone))
+
+            key_array = key.split(" ")
+            single_day = key_array[0]
+            time_key = datetime.strptime(single_day, "%Y-%m-%d")
+            next_day = time_key + timedelta(days=1)
+            next_day = next_day.__str__()
+
+
+
+            line_div.add_behavior( {
+                'type': 'click_up',
+                'cbjs_action': '''
+
+                var class_name = "tactic.ui.panel.FastTableLayoutWdg";
+                var popup_kwargs = {
+                    "search_type": "sthpw/milestone",
+                    "expression": "@SOBJECT(sthpw/milestone['due_date', '>=','%s']['due_date', '<=','%s'])"
+                    };
+
+                spt.panel.load_popup("Add Milestone", class_name, popup_kwargs);
+
+
+                    //"expression": "@SOBJECT(sthpw/milestone['due_date', 'moo'])"
+
+                ''' % (single_day, next_day)
+            } )
+
+
+        else:
+            num_milestone = 0
+            icon = IconWdg("Milestones", IconWdg.PLUS_ADD)
+            line_div.add(icon)
+            #line_div.add_style("opacity: 0.85")
+            line_div.add("Add milestone<br/>")
+            line_div.add_style("cursor: pointer;")
+        
+
+            line_div.add_behavior( {
+                'type': 'click_up',
+                'cbjs_action': '''
+                var server = TacticServerStub.get();
+                var date_div = bvr.src_el.getParent("td");
+                var date = date_div.firstChild.firstChild.textContent;
```

Comment:
need to use getParent or getElement to locate the date value.. This is too unreliable